timelock-agent/src/main/java/com/palantir/timelock/paxos/PaxosRemotingUtils.java

```diff
@@ -66,8 +64,8 @@ public static List<String> addProtocols(ClusterConfiguration cluster, List<Strin
 
     public static URL convertAddressToUrl(ClusterConfiguration cluster, String address) {
         try {
-            return new URIBuilder(addProtocol(cluster, address)).build().toURL();
-        } catch (MalformedURLException | URISyntaxException e) {
+            return new URL(addProtocol(cluster, address));
```

Comment:
In general you should probably test-drive this somewhere, the leniency and strictness of these parsers might differ. Although you'd think there's maybe some spec for URLs...

Reply:
Thanks, will check out dialogue and test this as well.

I checked the code and it's not immediately clear the parsers are equivalent.

The reason I initially went with the URIBuilder was that it was recommended in the Jakarta doc. I assumed it was considered a drop-in replacement for the purposes we care about.

Reply:
https://sourcegraph.palantir.build/search?q=context%3Aglobal+new+URL%28+repo%3Adialogue&patternType=standard&sm=1&groupBy=repo dialogue does use the URLs, will still test-drive!

Reply:
>The reason I initially went with the URIBuilder was that it was recommended in the Jakarta doc. I assumed it was considered a drop-in replacement for the purposes we care about.

Yep that's valid advice when you actually do anything with these builders, but it looks to me like this code was just a little bit unnecessary.